### PR TITLE
build: CI supports .NET 8

### PR DESCRIPTION
**Pull request by amanda-tarafa**
And KMS samples and tests are migrated to .NET 8 to confirm things work.

### CODE REVIEW

**Review by jskeet — DISMISSED** [review]
Yup, this looks encouraging.

**Comment by amanda-tarafa** [comment]
Windows ran KMS tests in .NET, but Linux didn't. I'm looking at this.

**Comment by amanda-tarafa** [comment]
Both Linux and Windows are now targetting .NET 8 for running KMS tests.
I'm making a few more changes that I will push shortly and we should be good.

**Comment by amanda-tarafa** [comment]
(I'm looking at these failures)

File: bigtable/api/Snippets/Reads.cs [hunk 1]
```diff
@@ -137,7 +144,10 @@ public string readRowRanges(string projectId = "YOUR-PROJECT-ID", string instanc
             ReadRowsStream readRowsStream = bigtableClient.ReadRows(tableName, rowSet);
```
Comment (jskeet) [inline bigtable/api/Snippets/Reads.cs:144]: We should probably have using statements for all of these streams...

File: bigtable/api/Snippets/Writes/WriteBatch.cs [hunk 2]
```diff
@@ -37,7 +37,7 @@ public class WriteBatch
 
             TableName tableName = new TableName(projectId, instanceId, tableId);
             BigtableVersion timestamp = new BigtableVersion(DateTime.UtcNow);
-            String COLUMN_FAMILY = "stats_summary";
+            string COLUMN_FAMILY = "stats_summary";
```
Comment (jskeet) [inline bigtable/api/Snippets/Writes/WriteBatch.cs:40]: As a follow-up we should probably rename this and make it a const (even if it's still as a local variable)

**Review by jskeet — APPROVED** [review]

File: bigtable/api/Snippets/Reads.cs (hunk 1 above)
Comment (amanda-tarafa) [reply bigtable/api/Snippets/Reads.cs:144]: Oh didn't see this before merging, but yep, I'll do that.

File: bigtable/api/Snippets/Writes/WriteBatch.cs (hunk 2 above)
Comment (amanda-tarafa) [reply bigtable/api/Snippets/Writes/WriteBatch.cs:40]: Yes, I think we shoulc rewrite a lot of these anyway. Let's what we can get done medium term.

**Review by amanda-tarafa — COMMENTED** [review]
Sorry, I merged before seeing your comments. I'll send a PR with usings for the streams shortly.

**Comment by jskeet** [comment]
> Sorry, I merged before seeing your comments.

That's fine, I'd intended them to be "for future refinement" anyway :)

File: bigtable/api/Snippets/Reads.cs (hunk 1 above)
Comment (amanda-tarafa) [reply bigtable/api/Snippets/Reads.cs:144]: ReadRowStream is not disposable, I've created b/352471768.

**Pull request merged**